fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -844,4 +844,59 @@ public CommandProcessingResult reActivateClient(Long entityId, JsonCommand comma
                 .withEntityId(entityId) //
                 .build();
     }
-}
\ No newline at end of file
+
+	@Override
+	public CommandProcessingResult undoRejection(Long entityId, JsonCommand command) {
+		final AppUser currentUser = this.context.authenticatedUser();
+		this.fromApiJsonDeserializer.validateUndoRejection(command);
+
+		final Client client = this.clientRepository.findOneWithNotFoundDetection(entityId);
+		final LocalDate undoRejectDate = command
+				.localDateValueOfParameterNamed(ClientApiConstants.reopenedDateParamName);
+
+		if (!client.isRejected()) {
```

Comment:
I do not think moving the validation to validator is absolutely required here. We are still avoiding entity creation, which is a good benefit.